modules/trusted-firmware-m/tfm_boards/common/tfm_hal_platform.c

```diff
@@ -103,14 +104,17 @@ static void maybe_log_for_gpio_port(uint32_t gpio_port, uint32_t secure_pin_mask
 		}
 	}
 }
+#endif
 
 static void log_pin_security_configuration(void)
 {
-#if CONFIG_NRF_GPIO0_PIN_MASK_SECURE || CONFIG_NRF_GPIO1_PIN_MASK_SECURE
-
+#ifdef CONFIG_NRF_GPIO0_PIN_MASK_SECURE
 	maybe_log_for_gpio_port(0, CONFIG_NRF_GPIO0_PIN_MASK_SECURE);
+#endif
+#ifdef CONFIG_NRF_GPIO1_PIN_MASK_SECURE
 	maybe_log_for_gpio_port(1, CONFIG_NRF_GPIO1_PIN_MASK_SECURE);
-#else
+#endif
+#if !defined(CONFIG_NRF_GPIO0_PIN_MASK_SECURE) && !defined(CONFIG_NRF_GPIO1_PIN_MASK_SECURE)
 	SPMLOG_INFMSG("All pins have been configured as non-secure\r\n");
```

Comment:
using #if instead of ifdef was intentional as I want
this print when we have

#define CONFIG_NRF_GPIO0_PIN_MASK_SECURE 0